src/opustrainer/trainer.py

```diff
@@ -24,6 +28,101 @@ def ignore_sigint():
     """
     signal.signal(signal.SIGINT, signal.SIG_IGN)
 
+# Placeholders, the Chinese edition, with some hardcoded values
+
+md = MosesDetokenizer(lang='zh')
+
+def get_placeholding_candidates(align_line: str) -> List[Tuple[int, int]]:
+    """Filters out multiple alignment targets so that we can definitely get a one-to-one replacement"""
+    def tuplify(pair: str) -> Tuple[int, int]:
+        s, t = pair.split('-')
+        return (int(s), int(t))
+    # Create the two src-trg and trg-src sets
+    src_trg: List[Tuple[int, int]] = [tuplify(i) for i in align_line.split()]
```

Comment:
I normally don't add types in places where it is pretty clear what they will be from the assignment itself. Most editors will be able to derive them here, and most humans do to? Up to you, but don't do the extra work for me 😄